### ISSUE TITLE
A question about 3DRegNet loss

### ISSUE THREAD

**Issue by QWTforGithub** [XuyangBai/PointDSC#14]
Sorry, I closed the last question by mistake. I have one last question about 3DRegNet. In your article, you only used classified loss to train 3DRegNet, namely BEC loss of PointDSC, right?

**Comment by XuyangBai** [comment]
Yes. You can also play with the transformation loss, but I do not see improvement during my experiments.

https://github.com/XuyangBai/PointDSC/blob/b009d536ac10b570853833f2178397c154745da9/libs/loss.py#L12

https://github.com/XuyangBai/PointDSC/blob/b009d536ac10b570853833f2178397c154745da9/config.py#L43-L44

**Comment by QWTforGithub** [comment]
> Yes. You can also play with the transformation loss, but I do not see improvement during my experiments.
> 
> https://github.com/XuyangBai/PointDSC/blob/b009d536ac10b570853833f2178397c154745da9/libs/loss.py#L12
> 
> https://github.com/XuyangBai/PointDSC/blob/b009d536ac10b570853833f2178397c154745da9/config.py#L43-L44

Thank you very much!

**Comment by QWTforGithub** [comment]
> Yes. You can also play with the transformation loss, but I do not see improvement during my experiments.
> 
> https://github.com/XuyangBai/PointDSC/blob/b009d536ac10b570853833f2178397c154745da9/libs/loss.py#L12
> 
> https://github.com/XuyangBai/PointDSC/blob/b009d536ac10b570853833f2178397c154745da9/config.py#L43-L44

Thank you for your thoughtful reply last time. I have so far trained 3DRegNet built by PointDSC according to the method you gave me. But I got 92.54% registration recall. I modify the PointDSC by following these steps:
1) comment **self.blocks[f'NonLocal_layer_{i}'] = NonLocalBlock(num_channels)** 
2) release  **nn.InstanceNorm1d(num_channels)** 
3) the backward propagation code of loss is modified as follows:
            **loss = self.metric_weight['ClassificationLoss'] * class_loss
            # loss = self.metric_weight['ClassificationLoss'] * class_loss + self.metric_weight['SpectralMatchingLoss'] * sm_loss           
            # if epoch > self.transformation_loss_start_epoch and self.metric_weight['TransformationLoss'] > 0.0:
            #     loss += self.metric_weight['TransformationLoss'] * trans_loss**
Did I miss something, or did I need to change any part of the test or validation code?

**Comment by XuyangBai** [comment]
Did you still use `class PointDSC`?  The 3DRegNet model contains the PointCN layers for feature extraction and an FC layer for classification, but PointDSC contained the SCNonlocal module for feature extraction and the NSM module for classification.  For your results, I suspect you just replace the SCNonLocal module with PointCNs, but still use the NSM proposed in PointDSC, then the network becomes a model mentioned in my ablation study (see  Row4 in Table 4 in my paper)
<img width="466" alt="截屏2022-01-30 下午3 27 26" src="https://user-images.githubusercontent.com/15166089/151690760-bc72e32d-fbc6-4c12-abc9-285d8c315169.png">

**Comment by QWTforGithub** [comment]
> 
Sorry, I'm clumsy.  I need to remove the NSM classification module from PointDSC and directly use the classification result of the following modules as the final classification result?   :
       **self.classification = nn.Sequential(
            nn.Conv1d(num_channels, 32, kernel_size=1, bias=True),
            nn.ReLU(inplace=True),
            nn.Conv1d(32, 32, kernel_size=1, bias=True),
            nn.ReLU(inplace=True),
            nn.Conv1d(32, 1, kernel_size=1, bias=True),
        )**

**Comment by XuyangBai** [comment]
Yes
